### PR TITLE
docs(mesh): update MeshGlobalRateLimit and MeshOPA targetref support table

### PR DESCRIPTION
**Pull request by slonka**
### Description

Change the targetRef support table since 2.6 it has a different layout.

**Listener tags are not yet supported because the plugins need to be adjusted**
 

### Testing instructions

Preview link: 

### Checklist 

- [X] Review label added 
- [X] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

For example, if this change is for an upcoming 3.6 release, enclose your content in `{% if_version gte:3.6.x %} <content> {% endif_version %}` tags (or `if_plugin_version` tags for plugins). 

Use any of the following keys:
* `gte:<version>` - greater than or equal to a specific version
* `lte:<version>` - less than or equal to a specific version
* `eq:<version>` - exactly equal to a specific version

You can do the same for older versions.

fix: https://github.com/Kong/kong-mesh/issues/5342

### CODE REVIEW

File: app/_src/mesh/features/meshglobalratelimit.md [hunk 1]
```diff
@@ -108,6 +108,26 @@ TODO: document how to generate and use zone token on universal.-->
...
+| `targetRef`             | Allowed kinds                         |
+| ----------------------- | ------------------------------------- |
+| `targetRef.kind`        | `Mesh`, `MeshSubset`, `MeshService`   |
+| `from[].targetRef.kind` | `Mesh`                                |
+{% endtab %}
+
+{% tab targetRef Builtin Gateway %}
+| `targetRef`             | Allowed kinds                                             |
+| ----------------------- | --------------------------------------------------------- |
+| `targetRef.kind`        | `Mesh`, `MeshGateway`                                     |
```
Comment (slonka) [inline app/_src/mesh/features/meshglobalratelimit.md:123]: do not merge, `MeshGateway` needs to be added to validator

File: app/_src/mesh/features/meshglobalratelimit.md [hunk 2]
```diff
@@ -108,6 +108,26 @@ TODO: document how to generate and use zone token on universal.-->
 
 ## TargetRef support matrix
 
+{% if_version gte:2.6.x %}
```
Comment (michaelbeaumont) [inline app/_src/mesh/features/meshglobalratelimit.md:111]:
```suggestion
{% if_version gte:2.7.x %}
```

**Review by michaelbeaumont — CHANGES_REQUESTED** [review]

File: app/_src/mesh/features/meshglobalratelimit.md [hunk 3]
```diff
@@ -108,6 +108,26 @@ TODO: document how to generate and use zone token on universal.-->
...
+| ----------------------- | ------------------------------------- |
+| `targetRef.kind`        | `Mesh`, `MeshSubset`, `MeshService`   |
+| `from[].targetRef.kind` | `Mesh`                                |
+{% endtab %}
+
+{% tab targetRef Builtin Gateway %}
+| `targetRef`             | Allowed kinds                                             |
+| ----------------------- | --------------------------------------------------------- |
+| `targetRef.kind`        | `Mesh`, `MeshGateway`                                     |
+| `from[].targetRef.kind` | `Mesh`                                                    |
```
Comment (michaelbeaumont) [inline app/_src/mesh/features/meshglobalratelimit.md:124]:
```suggestion
| `to[].targetRef.kind` | `Mesh`                                                    |
```

**Comment by slonka** [comment]
Thanks for reviewing @michaelbeaumont 👍

**Review by michaelbeaumont — APPROVED** [review]

**Comment by cloudjumpercat** [comment]
Hi @slonka ! If you rebase this PR, that should fix the failing broken links test (another PR was merged recently that fixes the 404 errors on dev).

**Comment by slonka** [comment]
Hi @cloudjumpercat ! Sorry for the delay, I've just updated the branch, let's see if it passes now.

**Comment by slonka** [comment]
@cloudjumpercat smoke tests still don't pass - it errors out with "invalid url" :(

**Comment by cloudjumpercat** [comment]
@slonka Thanks! The smoke tests are expected to fail since this is from a fork, so that's okay. I'll review it today

**Review by cloudjumpercat — APPROVED** [review]
LGTM! I'll ignore the failing smoke tests since that's an expected failure since this is from a forked repo.

**Pull request merged**